### ISSUE TITLE
Sparkline: u64-only data type blocks scientific/ML use cases

### ISSUE THREAD

**Issue by ryanoneill** [ryanoneill/envision#362]
## Summary

Sparkline only accepts `Vec<u64>` data. It cannot represent:
- Floating-point values (loss: 0.0342, accuracy: 87.5%)
- Negative values (residuals, gradients, deltas)
- Sub-integer precision of any kind

This makes it unusable for ML training dashboards, scientific monitoring, financial data, or anything involving real-valued measurements.

## Current API

```rust
pub enum SparklineMessage {
    Push(u64),
    PushBounded(u64, usize),
    SetData(Vec<u64>),
    // ...
}
```

Every data path is `u64`. The rendering uses ratatui's `Sparkline` widget which also uses `u64` internally.

## Why this matters

MetricsDashboard embeds Sparklines for per-metric history trends. This means the entire MetricsDashboard widget inherits the u64 limitation — you can't show a sparkline of loss values (0.342 → 0.298 → 0.251) or learning rates (1e-3 → 5e-4 → 1e-4) in a metrics grid.

For a visualization library targeting data-intensive applications, this is a fundamental limitation that undermines the usefulness of two components (Sparkline and MetricsDashboard).

## Suggested approach

Accept `f64` and handle the conversion internally:
1. Store `Vec<f64>` in SparklineState
2. At render time, normalize to the `u64` range that ratatui expects: map `[min, max]` → `[0, u64_render_max]`
3. Keep the external API as `f64`, do the scaling transparently

This preserves full precision in the data while working within ratatui's rendering constraints. The normalization step is trivial math.

For negative values, shift the baseline so that `min` maps to 0. Optionally, render a visual baseline marker at the zero-crossing point.
